yb-voyager/cmd/exportData.go

```diff
@@ -272,6 +283,78 @@ func exportData() bool {
 	}
 }
 
+// required only for postgresql/yugabytedb since GetAllTables() returns all tables and partitions
+func addLeafPartitionsInTableList(tableList []*sqlname.SourceName) ([]*sqlname.SourceName, error) {
+	requiredForSource := source.DBType == "postgresql" || source.DBType == "yugabytedb"
+	if !requiredForSource {
+		return tableList, nil
+	}
+	// here we are adding leaf partitions in the list only in yb or pg because events in the 
+	// these dbs are referred with leaf partitions only but in oracle root table is main point of reference 
+	// for partitions and in Oracle fall-forward/fall-back case we are doing renaming using the config `ybexporter.tables.rename` in dbzm 
+	// when event is coming from YB for leaf partitions it is getting renamed to root_table 
+	// ex - customers -> cust_other, cust_part11, cust_part12, cust_part22, cust_part21 
+	// events from oracle - will have customers for all of these partitions 
+	// events from yb,pg will have `cust_other, cust_part11, cust_part12, cust_part22, cust_part21` out of these leaf partitions only 
+	// using the dbzm we are renaming these events coming from yb to root_table for oracle.
+	// not required for pg to rename them.
+	
+	modifiedTableList := []*sqlname.SourceName{}
+
+	//TODO: optimisation to avoid multiple calls to DB with one call in the starting to fetch TablePartitionTree map.
+
+	//TODO: test when we upgrade to PG13+ as partitions are handled with root table
+	//Refer- https://debezium.zulipchat.com/#narrow/stream/302529-community-general/topic/Connector.20not.20working.20with.20partitions
+	for _, table := range tableList {
+		rootTable, err := GetRootTableOfPartition(table)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get root table of partition %s: %v", table.Qualified.MinQuoted, err)
+		}
+		allLeafPartitions := GetAllLeafPartitions(table)
+		switch true {
+		case len(allLeafPartitions) == 0 && rootTable != table: //leaf partition
+			partitionsToRootTableMap[table.Qualified.Unquoted] = rootTable.Qualified.MinQuoted // Unquoted->MinQuoted map as debezium uses Unquoted table name 
```

Comment:
Shall we get rid of this global variable `partitionsToRootTableMap` ? We can construct that map and return it from this function. 

The caller can then do whatever they want with it. 
in this case the caller will:
1. store in msr. 
2. pass it along to `prepareDebeziumConfig()`.